diff --git a/pal/uefi_acpi/PalLib.inf b/pal/uefi_acpi/PalLib.inf
index 060b8af..00543e8 100644
--- a/pal/uefi_acpi/PalLib.inf
+++ b/pal/uefi_acpi/PalLib.inf
@@ -46,7 +46,6 @@
   src/pal_pmu.c
   src/pal_pcc.c
   src/pal_tpm2.c
-  src/pal_cxl.c
 
 [Packages]
   ArmPkg/ArmPkg.dec
diff --git a/val/ValLibRB.inf b/val/ValLibRB.inf
index c0db1de..56f38f8 100644
--- a/val/ValLibRB.inf
+++ b/val/ValLibRB.inf
@@ -64,7 +64,6 @@
   src/acs_pfdi.c
   src/acs_msc_error.c
   src/acs_tpm.c
-  src/acs_cxl.c
   driver/smmu_v3/smmu_v3.c
   driver/gic/gic.c
   driver/gic/acs_exception.c
