diff --git a/BaseTools/Scripts/GccBase.lds b/BaseTools/Scripts/GccBase.lds
index 3d2a97ddeb..c9e35af0c1 100644
--- a/BaseTools/Scripts/GccBase.lds
+++ b/BaseTools/Scripts/GccBase.lds
@@ -27,6 +27,7 @@ SECTIONS {
   . = PECOFF_HEADER_SIZE;
 
   .text : ALIGN(CONSTANT(COMMONPAGESIZE)) {
+    _textbsa = .;
     KEEP(*(.entry))
     *(.text .text.* .stub .gnu.linkonce.t.*)
     *(.rodata .rodata.* .gnu.linkonce.r.*)
diff --git a/ShellPkg/Library/UefiShellCEntryLib/UefiShellCEntryLib.c b/ShellPkg/Library/UefiShellCEntryLib/UefiShellCEntryLib.c
index 17e48443ac..ddf56f0cd6 100644
--- a/ShellPkg/Library/UefiShellCEntryLib/UefiShellCEntryLib.c
+++ b/ShellPkg/Library/UefiShellCEntryLib/UefiShellCEntryLib.c
@@ -16,6 +16,9 @@
 #include <Library/ShellCEntryLib.h>
 #include <Library/DebugLib.h>
 
+EFI_SYSTEM_TABLE *mySystemTable;
+EFI_HANDLE myImageHandle;
+
 /**
   UEFI entry point for an application that will in turn call the
   ShellAppMain function which has parameters similar to a standard C
@@ -51,6 +54,9 @@ ShellCEntryLib (
   EfiShellParametersProtocol = NULL;
   EfiShellInterface          = NULL;
 
+  mySystemTable = SystemTable;
+  myImageHandle = ImageHandle;
+ 
   Status = SystemTable->BootServices->OpenProtocol (
                                         ImageHandle,
                                         &gEfiShellParametersProtocolGuid,
diff --git a/ShellPkg/ShellPkg.dsc b/ShellPkg/ShellPkg.dsc
index 94d95de0eb..4a70e2857a 100644
--- a/ShellPkg/ShellPkg.dsc
+++ b/ShellPkg/ShellPkg.dsc
@@ -23,6 +23,9 @@
 !include MdePkg/MdeLibs.dsc.inc
 
 [LibraryClasses.common]
+  ValLib|ShellPkg/Application/sysarch-acs/val/ValLib.inf
+  PalLib|ShellPkg/Application/sysarch-acs/pal/uefi_acpi/PalLib.inf
+  LibCFlat|ShellPkg/Application/sysarch-acs/mem_test/LibCFlat.inf
   UefiApplicationEntryPoint|MdePkg/Library/UefiApplicationEntryPoint/UefiApplicationEntryPoint.inf
   UefiDriverEntryPoint|MdePkg/Library/UefiDriverEntryPoint/UefiDriverEntryPoint.inf
   UefiBootServicesTableLib|MdePkg/Library/UefiBootServicesTableLib/UefiBootServicesTableLib.inf
@@ -79,6 +82,7 @@
   # Build all the libraries when building this package.
   # This helps developers test changes and how they affect the package.
   #
+  ShellPkg/Application/sysarch-acs/apps/uefi/Mem.inf
   ShellPkg/Library/UefiShellLib/UefiShellLib.inf
   ShellPkg/Library/UefiShellAcpiViewCommandLib/UefiShellAcpiViewCommandLib.inf
   ShellPkg/Library/UefiShellCommandLib/UefiShellCommandLib.inf
